src/test/java/com/edmunds/rest/databricks/service/SecretsServiceTest.java

```diff
@@ -0,0 +1,110 @@
+        } catch (Throwable throwable) {
+            System.err.println("Failed to remove ACL secret. " + throwable.getLocalizedMessage());
+        }
+
+        try {
+            service.deleteSecret("test-scope", "test-secret");
+        } catch (Throwable throwable) {
+            System.err.println("Failed to remove secret. " + throwable.getLocalizedMessage());
+        }
+
+        service.deleteSecretScope("test-scope");
+    }
+
+    @Test
+    public void createSecretScopeTest() throws IOException, DatabricksRestException {
+        NewSecretScopeDTO newSecretScopeDTO = new NewDatabricksSecretScopeDTO();
+        newSecretScopeDTO.setScope("test-scope");
+        newSecretScopeDTO.setInitialManagePrincipal("users");
+        service.createSecretScope(newSecretScopeDTO);
+    }
+
+    @Test(dependsOnMethods = {"createSecretScopeTest"})
+    public void testSecretScopesList() throws IOException, DatabricksRestException {
+
+        List<SecretScopeDTO> secretScopesDTO = service.listSecretScopes();
+        assertNotNull(secretScopesDTO);
+        boolean isTestScopePresent = secretScopesDTO.stream().anyMatch(secretScopeDTO -> secretScopeDTO.getName().equals("test-scope"));
+        assertTrue(isTestScopePresent);
+    }
+
+    @Test(dependsOnMethods = {"createSecretScopeTest"})
+    public void createSecretTest() throws DatabricksRestException, IOException {
+        NewSecretDTO secretDTO = new NewSecretDTO();
+        secretDTO.setScope("test-scope");
+        secretDTO.setKey("test-secret");
+        secretDTO.setStringValue("test-secret-value");
+
+        service.createSecret(secretDTO);
+    }
+
+    @Test(dependsOnMethods = {"createSecretTest"})
+    public void testSecretList() throws DatabricksRestException, IOException {
+        List<SecretDTO> secrets = service.listSecrets("test-scope");
+        assertNotNull(secrets);
+
+        boolean isSecretPresent = secrets.stream().anyMatch(secretScopeDTO -> secretScopeDTO.getKey().equals("test-secret"));
+        assertTrue(isSecretPresent);
+    }
+
+//    @Test(dependsOnMethods = {"createSecretScopeTest"})
+//    public void createACLsSecretsTests() throws DatabricksRestException, IOException {
+//        CreateACLSecretDTO createACLSecretDTO = new CreateACLSecretDTO();
+//        createACLSecretDTO.setScope("test-scope");
+//        createACLSecretDTO.setPrincipal("users");
+//        createACLSecretDTO.setPermission(ACLPermission.READ);
+//        service.createSecretACL(createACLSecretDTO);
+//    }
+//
+//    @Test(dependsOnMethods = {"createACLsSecretsTests"})
+//    public void listACLsSecretsTests() throws DatabricksRestException, IOException {
+//        List<ACLSecretDTO> secrets = service.listSecretACLs("test-scope");
+//        assertNotNull(secrets);
+//
+//        assertEquals(secrets.size(), 1);
+//
+//    }
+//
+//    @Test(dependsOnMethods = {"createACLsSecretsTests"})
+//    public void getACLsSecretsTests() throws DatabricksRestException, IOException {
```

Comment:
Do we want to keep these commented tests?

Reply:
yes, tests are fine, but we need a premium workspace to use them